### PR TITLE
DPE-4658 global-primary on endpoint

### PR DESCRIPTION
**Pull request by paulomach**
## Issue

1. Slow status update on switchover
1. Endpoint being empty breaks related router
1. cluster name clash workaround use meaningless hash

## Solution

* propagating switchover status change across cluster
* use global cluster set primary on standby endpoint
* use model_uuid to workaround cluster name clash (instead of generated uuid)

### CODE REVIEW

**Review by taurus-forever — APPROVED** [review]
Tnx!

**Comment by paulomach** [comment]
> > Endpoint being empty breaks related router
> 
> IMO, this shouldn't be fixed with
> 
> > use global cluster set primary on standby endpoint
> 
> because router will still fail on primary switchover if previous primary goes offline (e.g. region failure, network cut)—and this only hides the issue
> 
> but does this improve UX for consuming applications when async enabled?

The main point here is to not disrupt a running service due a switchover (failovers are a complete different beast).
Fact of matter is that we still don't have the full router on async history defined by Mohamed. 
Meanwhile this indeed avoids unnecessary disruption for end user.

File: lib/charms/mysql/v0/async_replication.py [hunk 1]
```diff
@@ -824,11 +842,12 @@ def _on_consumer_changed(self, event):  # noqa: C901
...
             if self.remote_relation_data["cluster-name"] == self.cluster_name:  # pyright: ignore
                 # this cluster need a new cluster name
+                # we append the model uuid, trimming to a max of 63 characters
                 logger.warning(
-                    "Cluster name is the same as the primary cluster. Appending generated value"
+                    "Cluster name is the same as the primary cluster. Appending model uuid"
                 )
                 self._charm.app_peer_data["cluster-name"] = (
-                    f"{self.cluster_name}{uuid.uuid4().hex[:4]}"
+                    f"{self.cluster_name}{self.model.uuid.replace('-', '')}"[:63]
```
[the remark this answers is not in the document]
Comment (paulomach) [reply lib/charms/mysql/v0/async_replication.py:850]: > could it be helpful to start with model name and then add uuid?

I don't see how this make things better. Bear in mind this is a rare edge case that does not interfere with operation.

> will users ever see this value?

In some logs and when querying cluster status with cluster-set option

**Comment by paulomach** [comment]
> is this solution helpful for non-router apps? or router only?

Specially helpful for non-router, but we always recommend router for async setup.

**Pull request merged**